dask/tests/test_delayed.py

```diff
@@ -164,6 +164,15 @@ def test_common_subexpressions():
     assert len(res.dask) == 3
 
 
+def test_delayed_optimize():
+    x = Delayed('b', {'a': 1,
+                      'b': (inc, 'a'),
+                      'c': (inc, 'b')})
```

Comment:
The only way I was able to make a graph out of `dask.delayed` objects that required culling was from manually building the graph and wrapping it in a delayed object. This is because delayed objects manually build their graphs out of the arguments to the functions, it's not possible (afaict) to build a graph using just the delayed api that doesn't depend on all enclosed keys. (@mrocklin do you remember what you were doing to run into this issue?)